src/main/java/com/dnd/runus/application/challenge/ChallengeService.java

```diff
@@ -23,28 +25,43 @@ public class ChallengeService {
 
     private final RunningRecordRepository runningRecordRepository;
 
-    public List<ChallengesResponse> getChallenges(long memberId) {
+    public List<ChallengeWithCondition> getChallenges(long memberId) {
         OffsetDateTime todayMidnight = LocalDate.now(SERVER_TIMEZONE_ID)
                 .atStartOfDay(SERVER_TIMEZONE_ID)
                 .toOffsetDateTime();
         OffsetDateTime yesterday = todayMidnight.minusDays(1);
 
-        List<ChallengesResponse> challengesResponses;
+        List<RunningRecord> runningRecords =
+                runningRecordRepository.findByMemberIdAndStartAtBetween(memberId, yesterday, todayMidnight);
+
+        List<ChallengeWithCondition> allChallengesWithConditions =
+                new ArrayList<>(challengeRepository.findAllActiveChallengesWithConditions());
+
         // 어제 기록이 없으면
-        if (!runningRecordRepository.hasByMemberIdAndStartAtBetween(memberId, yesterday, todayMidnight)) {
-            challengesResponses = challengeRepository.findAllIsNotDefeatYesterday().stream()
-                    .map(ChallengesResponse::from)
-                    .collect(Collectors.toList());
-        } else {
-            challengesResponses = challengeRepository.findAllChallenges().stream()
-                    .map(ChallengesResponse::from)
+        if (runningRecords.isEmpty()) {
+            allChallengesWithConditions = allChallengesWithConditions.stream()
+                    .filter(challengeWithCondition ->
+                            !challengeWithCondition.challenge().isDefeatYesterdayChallenge())
```

Comment:
`removeIf()`를 사용하면 변수 대입이 없어서 좋을 것 같아요

```suggestion
            allChallengesWithConditions.removeIf(challenge ->
                    challenge.challenge().isDefeatYesterdayChallenge()
            );
```

Reply:
`removeIf()`가 있군요! 좋은거 같아요!